internal/controller/ols_app_redis_reconciliator.go

```diff
@@ -40,22 +45,37 @@ func (r *OLSConfigReconciler) reconcileRedisServer(ctx context.Context, olsconfi
 }
 
 func (r *OLSConfigReconciler) reconcileRedisDeployment(ctx context.Context, cr *olsv1alpha1.OLSConfig) error {
-	deployment, err := r.generateRedisDeployment(cr)
+	desiredDeployment, err := r.generateRedisDeployment(cr)
 	if err != nil {
 		return fmt.Errorf("failed to generate OLS redis deployment: %w", err)
 	}
 
-	foundDeployment := &appsv1.Deployment{}
-	err = r.Client.Get(ctx, client.ObjectKey{Name: OLSAppRedisDeploymentName, Namespace: cr.Namespace}, foundDeployment)
+	existingDeployment := &appsv1.Deployment{}
+	err = r.Client.Get(ctx, client.ObjectKey{Name: OLSAppRedisDeploymentName, Namespace: cr.Namespace}, existingDeployment)
 	if err != nil && errors.IsNotFound(err) {
-		err = r.Create(ctx, deployment)
+		updateDeploymentAnnotations(desiredDeployment, map[string]string{
+			OLSConfigHashKey: r.stateCache[OLSConfigHashStateCacheKey],
+		})
+		updateDeploymentTemplateAnnotations(desiredDeployment, map[string]string{
+			OLSConfigHashKey: r.stateCache[OLSConfigHashStateCacheKey],
+		})
+		r.logger.Info("creating a new OLS redis deployment", "deployment", desiredDeployment.Name)
+		err = r.Create(ctx, desiredDeployment)
 		if err != nil {
 			return fmt.Errorf("failed to create OLS redis deployment: %w", err)
 		}
+		return nil
 	} else if err != nil {
 		return fmt.Errorf("failed to get OLS redis deployment: %w", err)
 	}
-	r.logger.Info("OLS redis deployment reconciled", "deployment", deployment.Name)
+
+	err = r.updateRedisDeployment(ctx, existingDeployment, desiredDeployment)
```

Comment:
This update of Redis deployment will trigger another round of reconciliation because the deployment's owner is the OLSConfig CR, update on it will trigger a update event on the owner.
We can add a update condition check before this. For example we check whether redis container arguments are changed.

Reply:
Thanks for caching this. From now onwards redis deployment will update only when there is an update in `RedisConfig`.